src/main/java/ch/njol/skript/expressions/ExprElement.java

```diff
@@ -175,6 +191,38 @@ protected T[] get(Event event) {
 		return elementArray;
 	}
 
+	@SuppressWarnings("unchecked")
+	private T @Nullable [] getFromQueue(Event event) {
+		SkriptQueue queue = (SkriptQueue) expr.getSingle(event);
+		if (queue == null)
+			return null;
+		int startIndex = 0, endIndex = 0;
+		if (this.startIndex != null) {
+			startIndex = this.startIndex.getOptionalSingle(event).orElse(1);
+			if (startIndex <= 0 && type != ElementType.RANGE)
+				return null;
+		}
+		if (this.endIndex != null) {
+			endIndex = this.endIndex.getOptionalSingle(event).orElse(1);;
+		}
+		return switch (type) {
+			case FIRST_ELEMENT -> CollectionUtils.array((T) queue.pollFirst());
+			case LAST_ELEMENT -> CollectionUtils.array((T) queue.pollLast());
+			case RANDOM -> CollectionUtils.array((T) queue.removeSafely(ThreadLocalRandom.current().nextInt(0, queue.size())));
+			case ORDINAL -> CollectionUtils.array((T) queue.removeSafely(startIndex - 1));
+			case TAIL_END_ORDINAL -> CollectionUtils.array((T) queue.removeSafely(queue.size() - startIndex));
+			case FIRST_X_ELEMENTS -> CollectionUtils.array((T[]) queue.removeRangeSafely(0, startIndex));
+			case LAST_X_ELEMENTS -> CollectionUtils.array((T[]) queue.removeRangeSafely(queue.size() - startIndex, queue.size()));
+			case RANGE -> {
+				boolean reverse = startIndex > endIndex;
+				T[] elements = CollectionUtils.array((T[]) queue.removeRangeSafely(Math.min(startIndex, endIndex) - 1, Math.max(startIndex, endIndex)));
```

Comment:
The min/max logic is already handled by the `removeRangeSafely` method it appears

Reply:
Yes but indices need to start at zero and `removeRangeSafely` doesn't handle that so I have to look for the starting number here in order to reduce it by one.